packages/eslint-plugin-next/src/rules/no-html-link-for-pages.ts

```diff
@@ -74,7 +74,18 @@ export = defineRule({
       }
       return fsExistsSyncCache[dir]
     })
-    if (foundPagesDirs.length === 0) {
+
+    const appDirs = rootDirs.map((dir) => [path.join(dir, 'app')]).flat()
```

Comment:
Looks like this should be the same as the `pagesDir` look up so that it detects `src/app` as well.

Reply:
This should be fixed now.